spring-cloud-stream-binder-rabbit/src/main/java/org/springframework/cloud/stream/binder/rabbit/RabbitMessageChannelBinder.java

```diff
@@ -269,13 +268,106 @@ protected MessageProducer createConsumerEndpoint(ConsumerDestination consumerDes
 		DefaultAmqpHeaderMapper mapper = DefaultAmqpHeaderMapper.inboundMapper();
 		mapper.setRequestHeaderNames(properties.getExtension().getHeaderPatterns());
 		adapter.setHeaderMapper(mapper);
+		ErrorInfrastructure errorInfrastructure = registerErrorInfrastructure(consumerDestination, group, properties);
+		if (properties.getMaxAttempts() > 1) {
+			adapter.setRetryTemplate(buildRetryTemplate(properties));
+			if (properties.getExtension().isRepublishToDlq()) {
+				adapter.setRecoveryCallback(errorInfrastructure.getRecoverer());
+			}
+		}
+		else {
+			adapter.setErrorChannel(errorInfrastructure.getErrorChannel());
+		}
 		adapter.afterPropertiesSet();
```

Comment:
Not related to this PR, but this `afterPropertiesSet()` is called from the `AbstractMessageChannelBinder`:
```
consumerEndpoint = createConsumerEndpoint(destination, group, properties);
consumerEndpoint.setOutputChannel(bridge);
if (consumerEndpoint instanceof InitializingBean) {
	((InitializingBean) consumerEndpoint).afterPropertiesSet();
}
```